File: code.java
public class DigitSum {
    public static int sumDigits(int n) {
        // Обработка отрицательных чисел
        if (n < 0) {
            return sumDigits(-n);
        }
        // Базовый случай рекурсии
        if (n < 10) {
            return n;
        }
        // Рекурсивный случай
        return n % 10 + sumDigits(n / 10);
    }
    
    public static void main(String[] args) {
        // Пример использования
        System.out.println(sumDigits(1234));  
        System.out.println(sumDigits(-456));  
    }
}

// Вывод программы:
// 10
// 15

File: code.cpp
#include <iostream>
using namespace std;

int sumDigits(int n) {
    // Базовый случай: если число меньше 10, возвращаем само число
    if (n < 10) {
        return n;
    }
    // Рекурсивный случай: последняя цифра + сумма цифр оставшейся части
    return n % 10 + sumDigits(n / 10);
}

int main() {
    cout << sumDigits(1236) << endl;  
    cout << sumDigits(9745) << endl;  
    cout << sumDigits(0) << endl;     
    return 0;
}

// Вывод программы:
// 12
// 25
// 0
